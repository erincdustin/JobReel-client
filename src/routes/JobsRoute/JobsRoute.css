@media screen and (min-width: 961px) {
  .job-search-results {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    grid-gap: 1em;
  }
  
  .job-results-container {
    position: relative;
    grid-column-start: 4;
    grid-column-end: 9;
    border-radius: 18px;
    height: 400px;
    position: relative;
    text-align: center;
  }

  .jobsRouteImage {
    position: relative;
    grid-column-start: 1;
    grid-column-end: 9;
  }

  .jobsRouteImage > img {
    height: 250px;
    width: 250px;
    position: absolute;
    top: -100px;
    right: -15%;
    z-index: -1;
  }
  
  #job-go-back {
    position: absolute;
    left: 0;
    top: 0;
    color:#3a3846;
  }
  
  #job-go-back:hover {
    color: #fc766a;
  }
  
  .job-results > h2 {
    text-align: center;
  }
  
  .job-results {
    max-height: 100%;
    overflow-y: auto;
    overflow-block: inherit;
  }
}

@media screen and (max-width: 960px) {
  .job-search-results {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    grid-gap: 1em;
  }

  .jobsRouteImage {
    position: relative;
    grid-column-start: 1;
    grid-column-end: 9;
  }

  .jobsRouteImage > img {
    height: 250px;
    width: 250px;
    position: absolute;
    top: -100px;
    right: -5%;
    z-index: -1;
  }
  
  .job-results-container {
    position: relative;
    grid-column-start: 1;
    grid-column-end: 9;
    border-radius: 18px;
    height: 400px;
    position: relative;
    text-align: center;
  }

  .topNav {
    grid-column-start: 1;
    grid-column-end: 9;
  }
  
  #job-go-back {
    position: absolute;
    left: 0;
    top: 0;
    color: #3a3846;
  }
  
  #job-go-back:hover {
    color: #fc766a;
  }
  
  .job-results > h2 {
    text-align: center;
  }
  
  .job-results {
    max-height: 100%;
    overflow-y: auto;
    overflow-block: inherit;
  }
}

@media (max-width: 960px) {
  .SideNav{
    display: none;
  }
}

@media (min-width: 960px) {
  .TopNav{
    display: none;
  }
}






