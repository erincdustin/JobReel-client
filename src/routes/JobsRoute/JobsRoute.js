import React, { Component } from 'react'
import JobReelContext from '../../context/JobReelContext';
import Job from '../../components/Job/Job'
import './JobsRoute.css';
import config from '../../config'
import TokenService from '../../services/token-service'
import GithubJob from '../../components/Job/GithubJob';
import SideNav from '../../components/SideNav/SideNav';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export default class JobsRoute extends Component {
    state = {
        search: null,
        savedJobUrls: {}
    }
    static contextType = JobReelContext

    componentDidMount() {
        const savedJobUrls = this.context.savedJobs.map(job => job.url);
        let savedJobUrlsObj = {};
        savedJobUrls.forEach(url => {
            savedJobUrlsObj[url] = url;
        })
        this.setState({ savedJobUrls: savedJobUrlsObj });
        const search = this.context.search
        setTimeout(() => {
            Promise.all([
                fetch(`${config.API_ENDPOINT}/jobs/authentic`, {
                    method: 'POST',
                    headers: {
                        'content-type': 'application/json',
                        'Authorization': `bearer ${TokenService.getAuthToken()}`,
                    },
                    body: JSON.stringify({
                        search
                    }),
                }),
                fetch(`${config.API_ENDPOINT}/jobs/github`, {
                    method: 'POST',
                    headers: {
                        'content-type': 'application/json',
                        'Authorization': `bearer ${TokenService.getAuthToken()}`,
                    },
                    body: JSON.stringify({
                        search
                    }),
                })
            ])
                .then(([res1, res2]) => Promise.all([res1.json(), res2.json()]))
                .then(([data1, data2]) => {
                    this.context.setGithubJobs(data2)
                    this.context.setAuthenticJobs(data1.listings.listing)
                });
        }, 500)
    }
   
    renderJobList() {
        const {gitHubJobs = [] } = this.context
        const {authenticJobs = [] } = this.context
        console.log(gitHubJobs)
        console.log(authenticJobs)
        const jobsListOne = gitHubJobs.map((job, i) => {
            return <GithubJob job={job} key={i} savedJobUrls={this.state.savedJobUrls}/>
        })
        const jobsListTwo = authenticJobs.map((job, i) => {
            return <Job job={job} company={job.company} type={job.type} location={job.company.name} key={i} savedJobUrls={this.state.savedJobUrls}/>
        })

        let joinedList = jobsListOne.concat(jobsListTwo);

        return (
            <div className='results'>
                {joinedList}
            </div>
        )
    }


    render() {
        return (
            <div className='job-search-results'>
                <div className='title'>
                    <h2>Jobs List</h2>
                </div>
                <SideNav />
                <div className='results-container'>
                    <Link  to={`/jobs`} alt="goBack">
                            <FontAwesomeIcon id='go-back' icon='times-circle' size='2x'/>
                    </Link>
                    {this.renderJobList()}
                </div>
                
                
            </div>
        )
    }

}