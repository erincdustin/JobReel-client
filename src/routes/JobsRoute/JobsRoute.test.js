import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import JobsRoute from './JobsRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PropTypes from 'prop-types';

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      savedJobs: [1, 2, 3]
    }
  })

  render() {
    return this.props.children;
  }
}

describe('JobsRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><JobsRoute/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('JobsRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
      <ContextProvider>
        <JobsRoute />
      </ContextProvider>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})