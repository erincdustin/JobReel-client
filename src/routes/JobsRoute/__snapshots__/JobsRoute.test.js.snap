// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobsRoute snapshot test renders the UI as expected 1`] = `<JobsRoute />`;
